File: app/static/js/sockets.js
const socket = io();

const btn_clicked = (id) => {
    console.log(id)
    socket.emit('onClick', id);
}

socket.on('onClick', (data) => {
    console.log(data)
    changeClass(data)
})

socket.on("user_connected", (data) => {
    console.log("user connected")
    console.log(data)
    for (let i = 0; i < data.length; i++) {
        changeClass(data[i])
    }
})



socket.on("reset", (data) => {
    console.log("user connected")
    resetAll()
})

const changeClass = (id) => {
    var element = document.getElementById(id);
    element.style.backgroundColor = "#f55";
}

const resetAll = () => {
    console.log("reset")
    for (let i = 1; i < 10; i++) {
        var element = document.getElementById(i);
        element.style.backgroundColor = "#444";
    }

}

File: app/websockets/sockets.py
from flask import request, session
from flask_socketio import SocketIO, emit

socketio = SocketIO(logger=True, cors_allowed_origins="*", async_mode="eventlet")

clicked = []


@socketio.on('connect')
def on_connect():
    print("user connected")
    emit('user_connected', clicked, broadcast=True)


@socketio.on('disconnect')
def on_disconnect():
    pass


@socketio.on('onClick')
def on_click(data):
    global clicked
    if data not in clicked:
        clicked.append(data)
        emit('onClick', data, broadcast=True, include_self=False)
    if len(clicked) > 8:
        clicked = []
        emit('reset', broadcast=True)

